pkg/network/domainspec/generators.go

```diff
@@ -254,96 +253,111 @@ type PasstLibvirtSpecGenerator struct {
 	vmi          *v1.VirtualMachineInstance
 }
 
+const (
+	passtLogFile      = "/var/run/kubevirt/passt.log" // #nosec G101
+	ifaceTypeUser     = "user"
+	ifaceBackendPasst = "passt"
+)
+
 func (b *PasstLibvirtSpecGenerator) Generate() error {
-	err := exec.Command("pgrep", "passt").Run()
-	if err == nil {
-		return fmt.Errorf("passt process is already running")
-	}
-	// remove passt interface from domain spec devices interfaces
-	foundDomainInterface := false
-	for i, iface := range b.domain.Spec.Devices.Interfaces {
-		if iface.Alias.GetName() == b.vmiSpecIface.Name {
-			b.domain.Spec.Devices.Interfaces = append(b.domain.Spec.Devices.Interfaces[:i], b.domain.Spec.Devices.Interfaces[i+1:]...)
-			foundDomainInterface = true
-			break
-		}
-	}
-	if !foundDomainInterface {
-		return fmt.Errorf("failed to find interface %s in vmi spec", b.vmiSpecIface.Name)
+	domainIface := LookupIfaceByAliasName(b.domain.Spec.Devices.Interfaces, b.vmiSpecIface.Name)
+	if domainIface == nil {
+		return fmt.Errorf("failed to find interface %s in domain spec", b.vmiSpecIface.Name)
 	}
 
-	ports := b.generatePorts()
-	args := append([]string{"--runas", "107", "-e"}, ports...)
-	cmd := exec.Command("/usr/bin/passt", args...)
-	cmd.SysProcAttr = &syscall.SysProcAttr{
-		AmbientCaps: []uintptr{unix.CAP_NET_BIND_SERVICE},
-	}
+	generatedIface := b.generateInterface(domainIface)
+	*domainIface = *generatedIface
 
-	// connect passt's stderr to our own stdout in order to see the logs in the container logs
-	var reader io.ReadCloser
-	reader, err = cmd.StderrPipe()
-	if err != nil {
-		log.Log.Reason(err).Error("failed to get passt stderr")
-		return err
-	}
 	go func() {
+		var logFile *os.File
+		for {
+			var err error
+			if logFile, err = os.Open(passtLogFile); err != nil && !errors.Is(err, os.ErrNotExist) {
+				log.Log.Reason(err).Errorf("failed to open file %s", passtLogFile)
+				return
+			} else if errors.Is(err, os.ErrNotExist) {
+				time.Sleep(time.Second)
+			} else if err == nil {
+				break
+			}
+		}
+		defer func() {
+			if closeErr := logFile.Close(); closeErr != nil {
+				log.Log.Reason(closeErr).Errorf("failed to close file: %s", passtLogFile)
+			}
+		}()
+
+		time.Sleep(time.Second)
```

Comment:
What is this for?

Reply:
This it to give chance for passt log messages populate.